### PR TITLE
Add common tasks to get ESXi hardware versions, guest ids and guest default config

### PR DESCRIPTION
**Pull request by keirazhang**
Example output:
```
TASK [Print ESXi supported hardware versions] ************************************************************************
ok: [localhost] => {
    "esxi_hardware_versions": [
        "10",
        "11",
        "12",
        "13",
        "14",
        "15",
        "16",
        "17",
        "18",
        "19",
        "20",
        "3",
        "4",
        "7",
        "8",
        "9"
    ]
}

TASK [Print ESXi default hardware version] ***************************************************************************
ok: [localhost] => {
    "esxi_default_hardware_version": "19"
}


TASK [Set default CPU number for VM with guest id windows9_64Guest] **************************************************
ok: [localhost]

TASK [Print VM default config with guest id windows9_64Guest] ********************************************************
ok: [localhost] => {
    "vm_default_config": {
        "boot_disk_controller": "lsilogicsas",
        "cdrom_controller_type": "sata",
        "cpu_cores_per_socket": "2",
        "cpu_number": 2,
        "cpu_socket_num": "1",
        "firmware": "efi",
        "memory_mb": "4096",
        "network_adapter_type": "e1000e",
        "secureboot_enabled": "true",
        "usb_controller": "usb3"
    }
}


TASK [Print hardware version 19 supported guest id on ESXi server] ***************************************************
ok: [localhost] => {
    "esxi_guest_ids": [
        "amazonlinux3_64Guest",
        "amazonlinux2_64Guest",
        "windows2019srvNext_64Guest",
        "windows2019srv_64Guest",
        "windows9Server64Guest",
        "windows8Server64Guest",
        "windows7Server64Guest",
        "winLonghorn64Guest",
        "winLonghornGuest",
        "winNetEnterprise64Guest",
        "winNetEnterpriseGuest",
        "winNetDatacenter64Guest",
        "winNetDatacenterGuest",
        "winNetStandard64Guest",
        "winNetStandardGuest",
        "winNetWebGuest",
        "winNetBusinessGuest",
        "windows9_64Guest",
        "windows9Guest",
        "windows8_64Guest",
        "windows8Guest",
        "windows7_64Guest",
        "windows7Guest",
        "winVista64Guest",
        "winVistaGuest",
        "winXPPro64Guest",
        "winXPProGuest",
        "win2000AdvServGuest",
        "win2000ServGuest",
        "win2000ProGuest",
        "winNTGuest",
        "win98Guest",
        "win95Guest",
        "win31Guest",
        "dosGuest",
        "vmwarePhoton64Guest",
        "crxPod1Guest",
        "rhel9_64Guest",
        "rhel8_64Guest",
        "rhel7_64Guest",
        "rhel6_64Guest",
        "rhel6Guest",
        "rhel5_64Guest",
        "rhel5Guest",
        "rhel4_64Guest",
        "rhel4Guest",
        "rhel3_64Guest",
        "rhel3Guest",
        "rhel2Guest",
        "sles16_64Guest",
        "sles15_64Guest",
        "sles12_64Guest",
        "sles11_64Guest",
        "sles11Guest",
        "sles10_64Guest",
        "sles10Guest",
        "sles64Guest",
        "slesGuest",
        "centos9_64Guest",
        "centos8_64Guest",
        "centos7_64Guest",
        "centos6_64Guest",
        "centos6Guest",
        "centos64Guest",
        "centosGuest",
        "debian11_64Guest",
        "debian11Guest",
        "debian10_64Guest",
        "debian10Guest",
        "debian9_64Guest",
        "debian9Guest",
        "debian8_64Guest",
        "debian8Guest",
        "debian7_64Guest",
        "debian7Guest",
        "debian6_64Guest",
        "debian6Guest",
        "debian5_64Guest",
        "debian5Guest",
        "debian4_64Guest",
        "debian4Guest",
        "opensuse64Guest",
        "opensuseGuest",
        "asianux9_64Guest",
        "asianux8_64Guest",
        "asianux7_64Guest",
        "asianux4_64Guest",
        "asianux4Guest",
        "asianux3_64Guest",
        "asianux3Guest",
        "fedora64Guest",
        "fedoraGuest",
        "oesGuest",
        "oracleLinux9_64Guest",
        "oracleLinux8_64Guest",
        "oracleLinux7_64Guest",
        "oracleLinux6_64Guest",
        "oracleLinux6Guest",
        "oracleLinux64Guest",
        "oracleLinuxGuest",
        "ubuntu64Guest",
        "ubuntuGuest",
        "coreos64Guest",
        "other5xLinux64Guest",
        "other5xLinuxGuest",
        "other4xLinux64Guest",
        "other4xLinuxGuest",
        "other3xLinux64Guest",
        "other3xLinuxGuest",
        "other26xLinux64Guest",
        "other26xLinuxGuest",
        "other24xLinux64Guest",
        "other24xLinuxGuest",
        "otherLinux64Guest",
        "otherLinuxGuest",
        "darwin21_64Guest",
        "darwin20_64Guest",
        "darwin19_64Guest",
        "darwin18_64Guest",
        "darwin17_64Guest",
        "darwin16_64Guest",
        "darwin15_64Guest",
        "darwin14_64Guest",
        "darwin13_64Guest",
        "darwin12_64Guest",
        "darwin11_64Guest",
        "darwin11Guest",
        "darwin10_64Guest",
        "darwin10Guest",
        "darwin64Guest",
        "darwinGuest",
        "freebsd13_64Guest",
        "freebsd13Guest",
        "freebsd12_64Guest",
        "freebsd12Guest",
        "freebsd11_64Guest",
        "freebsd11Guest",
        "freebsd64Guest",
        "freebsdGuest",
        "os2Guest",
        "netware6Guest",
        "netware5Guest",
        "solaris11_64Guest",
        "solaris10_64Guest",
        "solaris10Guest",
        "solaris9Guest",
        "solaris8Guest",
        "openServer6Guest",
        "openServer5Guest",
        "unixWare7Guest",
        "eComStation2Guest",
        "eComStationGuest",
        "otherGuest64",
        "otherGuest",
        "vmkernel65Guest",
        "vmkernel7Guest",
        "vmkernel6Guest",
        "vmkernel5Guest",
        "vmkernelGuest"
    ]
}
```

### CODE REVIEW

**Comment by keirazhang** [comment]
Output for ESXi 7.0.3:
common/esxi_get_hardware_versions.yml
```
TASK [debug] *********************************************************************************************************
ok: [localhost] => {
    "msg": [
        "ESXi server supported hardware versions: [4, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]",
        "ESXi server latest hardware version: 19",
        "ESXi server default hardware version: 19"
    ]
}
```
common/esxi_get_guest_ids.yml
```
TASK [Print ESXi server supported guest IDs on hardware version 19] ***************************************************
ok: [localhost] => {
    "esxi_guest_ids": [
        "amazonlinux3_64Guest",
        "amazonlinux2_64Guest",
        "windows2019srvNext_64Guest",
        "windows2019srv_64Guest",
        "windows9Server64Guest",
        "windows8Server64Guest",
        "windows7Server64Guest",
        "winLonghorn64Guest",
        "winLonghornGuest",
        "winNetEnterprise64Guest",
        "winNetEnterpriseGuest",
        "winNetDatacenter64Guest",
        "winNetDatacenterGuest",
        "winNetStandard64Guest",
        "winNetStandardGuest",
        "winNetWebGuest",
        "winNetBusinessGuest",
        "windows9_64Guest",
        "windows9Guest",
        "windows8_64Guest",
        "windows8Guest",
        "windows7_64Guest",
        "windows7Guest",
        "winVista64Guest",
        "winVistaGuest",
        "winXPPro64Guest",
        "winXPProGuest",
        "win2000AdvServGuest",
        "win2000ServGuest",
        "win2000ProGuest",
        "winNTGuest",
        "win98Guest",
        "win95Guest",
        "win31Guest",
        "dosGuest",
        "vmwarePhoton64Guest",
        "crxPod1Guest",
        "rhel9_64Guest",
        "rhel8_64Guest",
        "rhel7_64Guest",
        "rhel6_64Guest",
        "rhel6Guest",
        "rhel5_64Guest",
        "rhel5Guest",
        "rhel4_64Guest",
        "rhel4Guest",
        "rhel3_64Guest",
        "rhel3Guest",
        "rhel2Guest",
        "sles16_64Guest",
        "sles15_64Guest",
        "sles12_64Guest",
        "sles11_64Guest",
        "sles11Guest",
        "sles10_64Guest",
        "sles10Guest",
        "sles64Guest",
        "slesGuest",
        "centos9_64Guest",
        "centos8_64Guest",
        "centos7_64Guest",
        "centos6_64Guest",
        "centos6Guest",
        "centos64Guest",
        "centosGuest",
        "debian11_64Guest",
        "debian11Guest",
        "debian10_64Guest",
        "debian10Guest",
        "debian9_64Guest",
        "debian9Guest",
        "debian8_64Guest",
        "debian8Guest",
        "debian7_64Guest",
        "debian7Guest",
        "debian6_64Guest",
        "debian6Guest",
        "debian5_64Guest",
        "debian5Guest",
        "debian4_64Guest",
        "debian4Guest",
        "opensuse64Guest",
        "opensuseGuest",
        "asianux9_64Guest",
        "asianux8_64Guest",
        "asianux7_64Guest",
        "asianux4_64Guest",
        "asianux4Guest",
        "asianux3_64Guest",
        "asianux3Guest",
        "fedora64Guest",
        "fedoraGuest",
        "oesGuest",
        "oracleLinux9_64Guest",
        "oracleLinux8_64Guest",
        "oracleLinux7_64Guest",
        "oracleLinux6_64Guest",
        "oracleLinux6Guest",
        "oracleLinux64Guest",
        "oracleLinuxGuest",
        "ubuntu64Guest",
        "ubuntuGuest",
        "coreos64Guest",
        "other5xLinux64Guest",
        "other5xLinuxGuest",
        "other4xLinux64Guest",
        "other4xLinuxGuest",
        "other3xLinux64Guest",
        "other3xLinuxGuest",
        "other26xLinux64Guest",
        "other26xLinuxGuest",
        "other24xLinux64Guest",
        "other24xLinuxGuest",
        "otherLinux64Guest",
        "otherLinuxGuest",
        "darwin21_64Guest",
        "darwin20_64Guest",
        "darwin19_64Guest",
        "darwin18_64Guest",
        "darwin17_64Guest",
        "darwin16_64Guest",
        "darwin15_64Guest",
        "darwin14_64Guest",
        "darwin13_64Guest",
        "darwin12_64Guest",
        "darwin11_64Guest",
        "darwin11Guest",
        "darwin10_64Guest",
        "darwin10Guest",
        "darwin64Guest",
        "darwinGuest",
        "freebsd13_64Guest",
        "freebsd13Guest",
        "freebsd12_64Guest",
        "freebsd12Guest",
        "freebsd11_64Guest",
        "freebsd11Guest",
        "freebsd64Guest",
        "freebsdGuest",
        "os2Guest",
        "netware6Guest",
        "netware5Guest",
        "solaris11_64Guest",
        "solaris10_64Guest",
        "solaris10Guest",
        "solaris9Guest",
        "solaris8Guest",
        "openServer6Guest",
        "openServer5Guest",
        "unixWare7Guest",
        "eComStation2Guest",
        "eComStationGuest",
        "otherGuest64",
        "otherGuest",
        "vmkernel65Guest",
        "vmkernel7Guest",
        "vmkernel6Guest",
        "vmkernel5Guest",
        "vmkernelGuest"
    ]
}
```
common/esxi_get_guest_config_options.yml
```
TASK [Print guest id centos8_64Guest config options on hardware version 19] ******************************************
ok: [localhost] => {
    "guest_config_options": {
        "default_cdrom_controller": "sata",
        "default_cpu_cores_per_socket": "1",
        "default_cpu_number": 1,
        "default_cpu_socket": "1",
        "default_disk_controller": "paravirtual",
        "default_disk_size_in_mb": "16384",
        "default_firmware": "efi",
        "default_memory_in_mb": "2048",
        "default_network_adapter": "vmxnet3",
        "default_secure_boot": "true",
        "default_usb_controller": "usb2",
        "guest_fullname": "CentOS 8 (64-bit)",
        "guest_id": "centos8_64Guest",
        "hardware_version": "vmx-19",
        "rec_persistent_memory": "8192",
        "rec_vram_kb": "8192",
        "support_disk_controller": [
            "paravirtual",
            "ide",
            "sata",
            "nvme"
        ],
        "support_ethernet_card": [
            "e1000e",
            "sriov",
            "vmxnet3",
            "pvrdma"
        ],
        "support_min_persistent_mem_mb": "4",
        "support_persistent_memory": "true",
        "support_secure_boot": "true",
        "support_tpm_20": "true",
        "support_usb_controller": [
            "usb2",
            "usb3"
        ]
    }
}
```

Output for ESXi 6.7U3:
common/esxi_get_hardware_versions.yml
```
TASK [debug] *********************************************************************************************************
ok: [localhost] => {
    "msg": [
        "ESXi server supported hardware versions: [4, 7, 8, 9, 10, 11, 12, 13, 14, 15]",
        "ESXi server latest hardware version: 15",
        "ESXi server default hardware version: 14"
    ]
}
```
common/esxi_get_guest_ids.yml
```

TASK [Print ESXi server supported guest IDs on hardware version 14] **************************************************
task path: /home/qiz/workspace/github/ansible-vsphere-gos-validation/common/esxi_get_guest_ids.yml:65
ok: [localhost] => {
    "esxi_guest_ids": [
        "amazonlinux2_64Guest",
        "windows9Server64Guest",
        "windows8Server64Guest",
        "windows7Server64Guest",
        "winLonghorn64Guest",
        "winLonghornGuest",
        "winNetEnterprise64Guest",
        "winNetEnterpriseGuest",
        "winNetDatacenter64Guest",
        "winNetDatacenterGuest",
        "winNetStandard64Guest",
        "winNetStandardGuest",
        "winNetWebGuest",
        "winNetBusinessGuest",
        "windows9_64Guest",
        "windows9Guest",
        "windows8_64Guest",
        "windows8Guest",
        "windows7_64Guest",
        "windows7Guest",
        "winVista64Guest",
        "winVistaGuest",
        "winXPPro64Guest",
        "winXPProGuest",
        "win2000AdvServGuest",
        "win2000ServGuest",
        "win2000ProGuest",
        "winNTGuest",
        "win98Guest",
        "win95Guest",
        "win31Guest",
        "dosGuest",
        "vmwarePhoton64Guest",
        "rhel8_64Guest",
        "rhel7_64Guest",
        "rhel6_64Guest",
        "rhel6Guest",
        "rhel5_64Guest",
        "rhel5Guest",
        "rhel4_64Guest",
        "rhel4Guest",
        "rhel3_64Guest",
        "rhel3Guest",
        "rhel2Guest",
        "sles15_64Guest",
        "sles12_64Guest",
        "sles11_64Guest",
        "sles11Guest",
        "sles10_64Guest",
        "sles10Guest",
        "sles64Guest",
        "slesGuest",
        "centos8_64Guest",
        "centos7_64Guest",
        "centos6_64Guest",
        "centos6Guest",
        "centos64Guest",
        "centosGuest",
        "debian10_64Guest",
        "debian10Guest",
        "debian9_64Guest",
        "debian9Guest",
        "debian8_64Guest",
        "debian8Guest",
        "debian7_64Guest",
        "debian7Guest",
        "debian6_64Guest",
        "debian6Guest",
        "debian5_64Guest",
        "debian5Guest",
        "debian4_64Guest",
        "debian4Guest",
        "opensuse64Guest",
        "opensuseGuest",
        "asianux8_64Guest",
        "asianux7_64Guest",
        "asianux4_64Guest",
        "asianux4Guest",
        "asianux3_64Guest",
        "asianux3Guest",
        "fedora64Guest",
        "fedoraGuest",
        "oesGuest",
        "oracleLinux8_64Guest",
        "oracleLinux7_64Guest",
        "oracleLinux6_64Guest",
        "oracleLinux6Guest",
        "oracleLinux64Guest",
        "oracleLinuxGuest",
        "ubuntu64Guest",
        "ubuntuGuest",
        "coreos64Guest",
        "other4xLinux64Guest",
        "other4xLinuxGuest",
        "other3xLinux64Guest",
        "other3xLinuxGuest",
        "other26xLinux64Guest",
        "other26xLinuxGuest",
        "other24xLinux64Guest",
        "other24xLinuxGuest",
        "otherLinux64Guest",
        "otherLinuxGuest",
        "darwin18_64Guest",
        "darwin17_64Guest",
        "darwin16_64Guest",
        "darwin15_64Guest",
        "darwin14_64Guest",
        "darwin13_64Guest",
        "darwin12_64Guest",
        "darwin11_64Guest",
        "darwin11Guest",
        "darwin10_64Guest",
        "darwin10Guest",
        "darwin64Guest",
        "darwinGuest",
        "freebsd12_64Guest",
        "freebsd12Guest",
        "freebsd11_64Guest",
        "freebsd11Guest",
        "freebsd64Guest",
        "freebsdGuest",
        "os2Guest",
        "netware6Guest",
        "netware5Guest",
        "solaris11_64Guest",
        "solaris10_64Guest",
        "solaris10Guest",
        "solaris9Guest",
        "solaris8Guest",
        "openServer6Guest",
        "openServer5Guest",
        "unixWare7Guest",
        "eComStation2Guest",
        "eComStationGuest",
        "otherGuest64",
        "otherGuest",
        "vmkernel65Guest",
        "vmkernel6Guest",
        "vmkernel5Guest",
        "vmkernelGuest"
    ]
}
```
common/esxi_get_guest_config_options.yml
```
TASK [Print guest id centos8_64Guest config options on hardware version 14] ******************************************
ok: [localhost] => {
    "guest_config_options": {
        "default_cdrom_controller": "sata",
        "default_cpu_cores_per_socket": 1,
        "default_cpu_number": 1,
        "default_cpu_socket": 1,
        "default_disk_controller": "paravirtual",
        "default_disk_size_in_mb": 16384,
        "default_firmware": "efi",
        "default_memory_in_mb": 2048,
        "default_network_adapter": "vmxnet3",
        "default_secure_boot": true,
        "default_usb_controller": "usb2",
        "guest_fullname": "CentOS 8 (64-bit)",
        "guest_id": "centos8_64Guest",
        "hardware_version": "vmx-14",
        "rec_persistent_memory": 8192,
        "rec_vram_kb": 8192,
        "support_disk_controller": [
            "lsilogic",
            "paravirtual",
            "lsilogicsas",
            "sata",
            "nvme"
        ],
        "support_ethernet_card": [
            "vmxnet3",
            "e1000e",
            "sriov"
        ],
        "support_min_persistent_mem_mb": 4,
        "support_persistent_memory": true,
        "support_secure_boot": true,
        "support_tpm_20": false,
        "support_usb_controller": [
            "usb2",
            "usb3"
        ]
    }
}
```

**Comment by keirazhang** [comment]
@Tomorrow9 It seems support_disk_controller and support_ethernet_card get by community.vmware.vmware_vm_config_option don't include 'ide' and 'pvrdma'. Can you help to check the module? Thanks!

File: common/esxi_get_hardware_versions.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Get all supported hardware versions on ESXi server
+# Return:
+#   esxi_hardware_versions: A list of all supported hardware versions on ESXi server
+#   esxi_default_hardware_version: The deafult hardware version on the ESXi server
```
Comment (Tomorrow9) [inline common/esxi_get_hardware_versions.yml:7]: ESXi server latest hardware version is still a return value?

File: common/esxi_get_guest_ids.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Get all supported guest ids for a hardware version on ESXi server
+# Parameter:
+#   esxi_hardware_version: The hardware version on ESXi server, e.g. 13, 14, 15, etc.
+# Return:
+#   esxi_guest_ids: A list of all supported guest ids for hardware version
+
+- block:
```
Comment (Tomorrow9) [inline common/esxi_get_guest_ids.yml:10]: I think here `esxi_hardware_version` can be a required parameter must to be set when calling this task, and no need to call `esxi_get_hardware_versions.yml` one more time.

File: common/esxi_get_guest_ids.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    password: "{{ vsphere_host_user_password }}"
+    validate_certs: "{{ validate_certs | default(False) }}"
+    datacenter: "{{ vsphere_host_datacenter }}" 
+    esxi_hostname: "{{ esxi_hostname }}"
+    get_guest_os_ids: True
+    hardware_version: "vmx-{{ esxi_hardware_version }}"
+  register: get_guest_os_ids_result
+  ignore_errors: True
+
+- debug: var=get_guest_os_ids_result.instance
```
Comment (Tomorrow9) [inline common/esxi_get_guest_ids.yml:34]: debug task missing `name` and `when`.

File: common/esxi_get_guest_config_options.yml [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Get default VM config options for a specific guest id
+# Parameter
+#   guest_id: guest id
+#   esxi_hardware_version: ESXi hardware version
+# Return:
+#   guest_config_options: The guest ID config options for a hardware version
+
+- block:
```
Comment (Tomorrow9) [inline common/esxi_get_guest_config_options.yml:11]: I think here esxi_hardware_version can be a required parameter must to be set when calling this task, and no need to call esxi_get_hardware_versions.yml one more time.

File: common/esxi_get_guest_config_options.yml [hunk 5]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2021 VMware, Inc.
```
Comment (Tomorrow9) [inline common/esxi_get_guest_config_options.yml:1]: copyright is 2022.

File: common/esxi_get_guest_config_options.yml [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+  set_fact:
+    guest_config_options: "{{ guest_config_options | combine({'default_cdrom_controller': 'ide'}) }}"
+  when: >
+    (guest_config_options.default_cdrom_controller is undefined) or
+    (not guest_config_options.default_cdrom_controller)
+
+- name: "Print guest id {{ guest_id }} config options on hardware version {{ esxi_hardware_version }}"
+  debug: var=guest_config_options
+
+# Example output:
```
Comment (Tomorrow9) [inline common/esxi_get_guest_config_options.yml:274]: Example output can be put at the beginning like in other tasks.

File: common/esxi_get_guest_config_options.yml [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+          }}
+
+    # Create a temp file to store config options
+    - include_tasks: create_temp_file_dir.yml
+    - name: "Set fact of temp config option file"
+      set_fact:
+        tmp_config_option_file: "{{ tmp_path }}"
+
+    - name: "Fetch config option content from ESXi server"
+      shell: "cat {{ vm_config_option_esx_hw }}"
```
Comment (Tomorrow9) [inline common/esxi_get_guest_config_options.yml:107]: Can this file be transferred to local host directly?

File: common/esxi_get_guest_config_options.yml [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    - guest_config_options.default_cpu_socket | int >= 1
+
+- name: "Set default CPU number for guest id {{ guest_id }}"
+  set_fact:
+    guest_config_options: "{{ guest_config_options | combine({'default_cpu_number': 1}) }}"
+  when: >
+    (guest_config_options.default_cpu_number is undefined) or
+    (not guest_config_options.default_cpu_number)
+
+- name: "Set default USB controller for guest id {{ guest_id }}"
```
Comment (Tomorrow9) [inline common/esxi_get_guest_config_options.yml:257]: I think we can leave these as what they are retrieved, handle it later when they're used, fail or set a default value or ignore if it's empty or not set.

File: common/esxi_get_hardware_versions.yml (hunk 1 above)
Comment (keirazhang) [reply common/esxi_get_hardware_versions.yml:7]: Fixed.

File: common/esxi_get_guest_ids.yml (hunk 2 above)
Comment (keirazhang) [reply common/esxi_get_guest_ids.yml]: Add an assert task for checking esxi_hardware_version

File: common/esxi_get_guest_config_options.yml (hunk 5 above)
Comment (keirazhang) [reply common/esxi_get_guest_config_options.yml]: Copyright was updated.

File: common/esxi_get_guest_config_options.yml (hunk 7 above)
Comment (keirazhang) [reply common/esxi_get_guest_config_options.yml]: Replaced with fetch task

File: common/esxi_get_guest_config_options.yml (hunk 8 above)
Comment (keirazhang) [reply common/esxi_get_guest_config_options.yml]: Removed tasks for setting default values if they're not retrieved.

File: common/esxi_get_guest_config_options.yml (hunk 6 above)
Comment (keirazhang) [reply common/esxi_get_guest_config_options.yml]: Moved example output at the beginning.

File: common/esxi_get_guest_config_options.yml [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+    - name: "Initialize VM default config options"
+      set_fact:
+        guest_config_options: >-
+          {{
+            vm_config_name.values() | zip_longest([]) 
+            | items2dict(key_name=0, value_name=1)
+          }}
+
+    # Create a temp file to store config options
+    - include_tasks: create_temp_file_dir.yml
```
Comment (Tomorrow9) [inline common/esxi_get_guest_config_options.yml:139]: There is a cache folder `local_cache` created at the beginning for downloaded files.

Comment (keirazhang) [reply common/esxi_get_guest_config_options.yml:139]: Thanks! I just added tmp_dir variable for this task.

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**